server/controllers/campaigns.js

```diff
@@ -70,3 +70,57 @@ export function getAllCampaigns(req, res) {
     });
 }
 
+
+function validCampaignStatusUpdate(prevStatus, currStatus) {
+  const validCampaignStatusTransitions = {
+    draft: 'active',
+    active: ['pause', 'completed'],
+    pause: ['active', 'completed'],
+    completed: 1
+  };
+  const validStatusOptions = Object.keys(validCampaignStatusTransitions);
+  let validTransitionForStatus;
+  if (prevStatus === 'completed') {
+    throw new Error('Campaigns with status set to "completed" may not be changed.');
+  }
+  if (!validCampaignStatusTransitions[currStatus]) {
+    throw new Error(`Status update to: ${currStatus} is invalid, valid status options: ${validStatusOptions.join(', ')}.`);
+  }
+  if (prevStatus === 'active' || prevStatus === 'pause') {
```

Comment:
You can get rid of dealing with draft as a special case by maybe putting `draft: ['active'] in line 76? That way we just deal with all 3 statuses with list values? What do you think? Lets us clean up the logic in here a bit?